### ISSUE TITLE
RTE Architecture

### ISSUE THREAD

**Issue by adriatic** [adriatic/aureliatools#3]
#### Let's take the strict Aurelian approach:

- **RTE client** is the Aurelia app using as few Aurelia Framework components as it takes to render the desktop application quality to **RTE engine** which in turn is a "headless" NodeJS application. The RTE engine has nearly infinite scalability and uses MongoDB to persist all user data. 

- All communication between RTE clients and RTE Engine use **[JSONP](http://json-jsonp-tutorial.craic.com/index.html)** or "better" protocol.

- As it is the case with any serious SPA app, all authentication and authorization is handles on the Identity Server (which is a different entity than the RTE Engine)

- The Code Editor lives in the browser

- The Run command sends the code to the RTE engine and waits for the the results (no real benefit from doing this asynchronously and it gets enormously more complex

- ... more to come

**Comment by JeroenVinke** [comment]
So the RTE client is the app where users can create samples in?

**Comment by JeroenVinke** [comment]
![image](https://cloud.githubusercontent.com/assets/2189477/12820679/6d66e588-cb60-11e5-9362-7ea948e9d67d.png)

I'd rather let the RTE engine communicate to Github directly, OAuth is hard complex enough as is

**Comment by adriatic** [comment]
> I'd rather let the RTE engine communicate to Github directly
 I agree with the caveat that I proposed to have Github authentication implemented in a pass-through shell, allowing eventual expansion at a later time (if necessary)
